java/ql/test/query-tests/security/CWE-023/semmle/tests/PartialPathTraversalTest.java

```diff
@@ -225,6 +232,18 @@ void foo24(File dir, File parent) throws IOException {
         }
     }
 
+    boolean foo25(String child) {
+        return child.startsWith(basePath);
+    }
```

Comment:
IMHO, this case should be getting flagged, but it doesn't look like `DataFlow::Configuration` is seeing the flow from the constructor to this call. @smowton do you have any insights into what's going on here?

Reply:
`PartialPathTraversalTest` isn't instantiated anywhere, nor `foo25` is called. So there's no dataflow edge connecting the implicit `this` qualifier of `basePath` in the constructor to the instance parameter of `foo25`, which is what you want.

You'd see flow if you added those calls. Something like:

```java
private static void test() {
  PartialPathTraversalTest t = new PartialPathTraversalTest(null);
  t.foo25();
}
```

Reply:
Awesome!  Much appreciated!